src/rust/engine/protos/protos/pants/cache.proto

```diff
@@ -17,6 +17,21 @@ message CacheKey {
   build.bazel.remote.execution.v2.Digest digest = 2;
 }
 
+message DependencyInferenceRequest {
+  build.bazel.remote.execution.v2.Digest input_file_digest = 1;
+  optional JavascriptInferenceMetadata metadata = 2;
```

Comment:
Since only one of these fields will be set at a time, this should likely be a "union" using: https://protobuf.dev/programming-guides/proto3/#oneof